python/doc_generator.py

```diff
@@ -47,13 +75,51 @@ def _get_transform_markdown(transform_yaml: Dict,
         md.h2('Parameters'),
         md.table(transform_yaml['arguments']),
         md.h2('Example'),
+        md.text(DATASET_PREVIEW_TEXT) if 'preview-data' in transform_yaml else None,
+        md.python_code(DATASET_PREVIEW_CODE) if 'preview-data' in transform_yaml else None,
+        t1,# md.text(transform_yaml['preview-data']) if 'preview-data' in transform_yaml else None,
+        md.text(TRANSFORMED_DATA_PREVIEW_TEXT) if 'preview-data' in transform_yaml else None,
         md.python_code(transform_yaml['example_code']),
+        t2,# md.text(transform_yaml['post-transform-data']) if 'preview-data' in transform_yaml else None,
         md.h2('Source Code'),
         md.github_url(transform_type_dir_name, transform_name),
         '',
     ]
-    return '\n\n'.join(markdown_elements)
+    return '\n\n'.join(filter(lambda x: x!=None, markdown_elements))
 
+def _generate_dataset_markdowns(
+    rasgo: Any,
+    dataset_id: int,
+    output_cols: List[str],
+    transform_details: Dict
+) -> List[str]:
+    """Creates markdown representation of both a pre-transform Rasgo Dataset
+    and a transformed Rasgo Dataset
+
+    Arguments:
+        :dataset_id: int: A Rasgo Community Dataset ID matching the columns named 
+        in the `example_code` section of the YAML doc
+        :output_cols: List[str]: A list of column names to include in the output
+        YAML representation
+        :transform_args: dict: TODO what do we want here? How to build the transform from args
+    """
+    ds = rasgo.get.dataset(dataset_id)
+    ds_md = ds.preview()[output_cols].to_markdown()
+
+    ds1 = ds.transform(
+        transform_name=transform_details["transform-name"],
+        arguments=transform_details["transform-args"]
+    )
+    ds1_md = ds1.preview().to_markdown()
+
+    return[ds_md, ds1_md]
 
 if __name__ == '__main__':
-    save_transform_docs()
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument('pyrasgo-api-key', action='store')
+    args = arg_parser.parse_args()
+
+
+    save_transform_docs(
+        rasgo_api_key=getattr(args, 'pyrasgo-api-key')
+    )
```

Comment:
Would be good to allow the user to specify which rasgo domain to hit (local vs staging vs prod).

Check out `python/publish_transforms.py` to see how we handle this. You could add the arg parse below and then use the function `utils.set_rasgo_domain_env(rasgo_domain)`

```python
arg_parser.add_argument('-d', '--rasgo-domain',
                        action='store',
                        default='production',
                        choices=['local', 'staging', 'production'],
                        help="Rasgo Environment to connect to. Sets env var "
                             "'RASGO_DOMAIN' for PyRasgo url dispatching "
                             "(local, staging, or production).")
```

Reply:
yeah this makes sense, and I actually had it in the first pass, but is anything in this code dependent on the domain? Community dataset IDs will remain the same, and we're not writing anything out to a place where we need the domain to determine the path?